File: uis/client/web2/src/models/wallet.ts
export interface WalletJson {
  id?: string;
  wallet_id?: string;
  user_id?: string;
  userId?: string;
  account_number?: string;
  accountNumber?: string;
  balance?: number;
  available_balance?: number;
  availableBalance?: number;
  currency?: string;
  status?: string;
  created_at?: string;
  createdAt?: string;
  updated_at?: string;
  updatedAt?: string;
}

export class Wallet {
  id: string;
  userId: string;
  accountNumber: string;
  balance: number;
  availableBalance: number;
  currency: string;
  status: string;
  createdAt: Date;
  updatedAt?: Date;

  constructor(data: {
    id: string;
    userId: string;
    accountNumber: string;
    balance: number;
    availableBalance: number;
    currency: string;
    status: string;
    createdAt: Date;
    updatedAt?: Date;
  }) {
   this.id = data.id;
   this.userId = data.userId;
   this.accountNumber = data.accountNumber;
   this.balance = data.balance;
   this.availableBalance = data.availableBalance;
   this.currency = data.currency;
   this.status = data.status;
   this.createdAt = data.createdAt;
   this.updatedAt = data.updatedAt;
  }

  static fromJson(json: WalletJson): Wallet {
    return new Wallet({
      id: json.id ?? json.wallet_id ?? '',
      userId: json.user_id ?? json.userId ?? '',
      accountNumber: json.account_number ?? json.accountNumber ?? '',
      balance: Number(json.balance ?? 0),
      availableBalance: Number(json.available_balance ?? json.availableBalance ?? 0),
      currency: json.currency ?? 'NGN',
      status: json.status ?? 'active',
      createdAt: new Date(json.created_at ?? json.createdAt ?? new Date().toISOString()),
      updatedAt: json.updated_at || json.updatedAt ? new Date(json.updated_at ?? json.updatedAt!) : undefined,
    });
  }

  toJson(): WalletJson {
    return {
      id: this.id,
      user_id: this.userId,
      account_number: this.accountNumber,
      balance: this.balance,
      available_balance: this.availableBalance,
      currency: this.currency,
      status: this.status,
      created_at: this.createdAt.toISOString(),
      updated_at: this.updatedAt?.toISOString(),
    };
  }

  get formattedBalance(): string {
    return `₦${this.balance.toFixed(2).replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
  }

  get formattedAvailableBalance(): string {
    return `₦${this.availableBalance.toFixed(2).replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
  }
}

File: uis/client/mobile_app/lib/models/wallet.dart
class Wallet {
  final String id;
  final String userId;
  final String accountNumber;
  final double balance;
  final double availableBalance;
  final String currency;
  final String status;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Wallet({
    required this.id,
    required this.userId,
    required this.accountNumber,
    required this.balance,
    required this.availableBalance,
    required this.currency,
    required this.status,
    required this.createdAt,
    this.updatedAt,
  });

  factory Wallet.fromJson(Map<String, dynamic> json) {
    return Wallet(
      id: json['id'] ?? json['wallet_id'] ?? '',
      userId: json['user_id'] ?? json['userId'] ?? '',
      accountNumber: json['account_number'] ?? json['accountNumber'] ?? '',
      balance: (json['balance'] ?? 0).toDouble(),
      availableBalance: (json['available_balance'] ?? json['availableBalance'] ?? 0).toDouble(),
      currency: json['currency'] ?? 'NGN',
      status: json['status'] ?? 'active',
      createdAt: DateTime.parse(json['created_at'] ?? json['createdAt'] ?? DateTime.now().toIso8601String()),
      updatedAt: json['updated_at'] != null || json['updatedAt'] != null
          ? DateTime.parse(json['updated_at'] ?? json['updatedAt'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'account_number': accountNumber,
      'balance': balance,
      'available_balance': availableBalance,
      'currency': currency,
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }

  String get formattedBalance {
    return '₦${balance.toStringAsFixed(2).replaceAllMapped(RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))'), (Match m) => '${m[1]},')}';
  }

  String get formattedAvailableBalance {
    return '₦${availableBalance.toStringAsFixed(2).replaceAllMapped(RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))'), (Match m) => '${m[1]},')}';
  }
}
